「Create post」ワークフローに失敗する

https://github.com/ios-osushi/website/runs/6287369544?check_suite_focus=true

```
Run POST_FILENAME="006-20220509.md"
  POST_FILENAME="006-20220509.md"
  POST_FILE="./Content/posts/$POST_FILENAME"
  POST_DATE_BEFORE=20220509
  POST_YEAR=${POST_DATE_BEFORE:0:4}
  POST_MONTH=${POST_DATE_BEFORE:4:2}
  POST_DAY=${POST_DATE_BEFORE:6:2}
  POST_DATE=$POST_YEAR-$POST_MONTH-$POST_DAY
  touch $POST_FILE
  echo -e "---\ndate: $POST_DATE 09:00\ndescription: TBD\ntags: TBD\n---\n# 006 $POST_DATE\n\n## TBD\n\nTBD" > $POST_FILE
  git add $POST_FILE
  git commit -m "Create $POST_FILENAME"
  git push origin feature/006_20220509
  echo "::set-output name=post_filename::$POST_FILENAME"
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_TOKEN: ***
[feature/006_20220509 4295d93 Create 006-20220509.md
 1 file changed, 10 insertions(+)
 create mode 100644 Content/posts/006-20220509.md
To https://github.com/ios-osushi/website
 ! [rejected]        feature/006_20220509 -> feature/006_20220509 (fetch first)
error: failed to push some refs to 'https://github.com/ios-osushi/website'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
Error: Process completed with exit code 1.
```

自分で実行してみたら、↑のステップは成功したけど、次のステップで失敗した↓
https://github.com/ios-osushi/website/runs/6300125578?check_suite_focus=true

```
Run hub version
git version 2.36.0
hub version 2.14.2
Error creating pull request: Forbidden (HTTP 403)
GitHub Actions is not permitted to create or approve pull requests.
Error: Process completed with exit code 1.
```

メモ
`contents: write` にしないとブランチをプッシュできなかった
https://github.com/ios-osushi/website/runs/6300412234?check_suite_focus=true

```
Run POST_FILENAME="999-20220505.md"
  POST_FILENAME="999-20220505.md"
  POST_FILE="./Content/posts/$POST_FILENAME"
  POST_DATE_BEFORE=20220505
  POST_YEAR=${POST_DATE_BEFORE:0:4}
  POST_MONTH=${POST_DATE_BEFORE:4:2}
  POST_DAY=${POST_DATE_BEFORE:6:2}
  POST_DATE=$POST_YEAR-$POST_MONTH-$POST_DAY
  touch $POST_FILE
  echo -e "---\ndate: $POST_DATE 09:00\ndescription: TBD\ntags: TBD\n---\n# 999 $POST_DATE\n\n## TBD\n\nTBD" > $POST_FILE
  git add $POST_FILE
  git commit -m "Create $POST_FILENAME"
  git push origin feature/999_20220505
  echo "::set-output name=post_filename::$POST_FILENAME"
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_TOKEN: ***
[feature/999_20220505 7e5b75d] Create 999-20220505.md
 1 file changed, 10 insertions(+)
 create mode 100644 Content/posts/999-20220505.md
remote: Permission to ios-osushi/website.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/ios-osushi/website/': The requested URL returned error: 403
Error: Process completed with exit code 128.
```

`hub` から `gh` に変えてもエラーだなぁ
https://github.com/ios-osushi/website/runs/6300703162?check_suite_focus=true

```
Run gh --version
  gh --version
  gh pr create \
    --title "Create 999-20220505.md" \
    --body "## スクリーンショット\n\n| `999-20220505.md` |\n|:--:|\n|TBD|" \
    --base "main" \
    --head "feature/999_20220505" \
    --draft
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_TOKEN: ***
gh version 2.8.0 (2022-04-13)
https://github.com/cli/cli/releases/tag/v2.8.0
/home/runner/work/_temp/3de12cf0-a0c4-42c6-86ff-ba9810e81f86.sh: line 2: 999-20220505.md: command not found
pull request create failed: GraphQL: GitHub Actions is not permitted to create or approve pull requests (createPullRequest)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -6,6 +6,9 @@ on:
   pull_request:
     branches: [ main ]
 
+permissions:
+  contents: read
+
 jobs:
   build:
     runs-on: ubuntu-latest

--- a/.github/workflows/create_post.yml
+++ b/.github/workflows/create_post.yml
@@ -13,6 +13,7 @@ on:
         type: string
 
 permissions:
+  contents: write
   pull-requests: write
 
 jobs:

--- a/.github/workflows/create_post.yml
+++ b/.github/workflows/create_post.yml
@@ -12,9 +12,13 @@ on:
         required: true 
         type: string
 
+permissions:
+  pull-requests: write
+
 jobs:
   create-post:
     runs-on: ubuntu-latest
+
     steps:
     - uses: actions/checkout@v3
 

--- a/.github/workflows/create_post.yml
+++ b/.github/workflows/create_post.yml
@@ -65,4 +65,3 @@ jobs:
           --draft
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
- \

--- a/.github/workflows/create_post.yml
+++ b/.github/workflows/create_post.yml
@@ -59,9 +59,10 @@ jobs:
         gh --version
         gh pr create \
           --title "Create ${{ steps.create_post.outputs.post_filename }}" \
-          --body "## スクリーンショット\n\n| `${{ steps.create_post.outputs.post_filename }}` |\n|:--:|\n|TBD|"
+          --body "## スクリーンショット\n\n| `${{ steps.create_post.outputs.post_filename }}` |\n|:--:|\n|TBD|" \
           --base "${{ github.event.repository.default_branch }}" \
           --head "${{ steps.create_work_branch.outputs.work_branch_name }}" \
           --draft
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+ \

--- a/.github/workflows/create_post.yml
+++ b/.github/workflows/create_post.yml
@@ -56,9 +56,10 @@ jobs:
 
     - name: Create pull request
       run: |
-        hub version
-        hub pull-request \
-          --message "Create ${{ steps.create_post.outputs.post_filename }}" \
+        gh --version
+        gh pr create \
+          --title "Create ${{ steps.create_post.outputs.post_filename }}" \
+          --body "## スクリーンショット\n\n| `${{ steps.create_post.outputs.post_filename }}` |\n|:--:|\n|TBD|"
           --base "${{ github.event.repository.default_branch }}" \
           --head "${{ steps.create_work_branch.outputs.work_branch_name }}" \
           --draft

--- a/.github/workflows/create_post.yml
+++ b/.github/workflows/create_post.yml
@@ -59,7 +59,6 @@ jobs:
         gh --version
         gh pr create \
           --title "Create ${{ steps.create_post.outputs.post_filename }}" \
-          --body "## スクリーンショット\n\n| `${{ steps.create_post.outputs.post_filename }}` |\n|:--:|\n|TBD|" \
           --base "${{ github.event.repository.default_branch }}" \
           --head "${{ steps.create_work_branch.outputs.work_branch_name }}" \
           --draft

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -5,6 +5,9 @@ on:
     - cron: '0 23 * * 0'
   workflow_dispatch:
 
+permissions:
+  contents: read
+
 jobs:
   build:
     runs-on: ubuntu-latest

